src/main/java/com/cloudcoders/gestaca/persistance/CourseDAOImpl.java

```diff
@@ -21,30 +22,33 @@ public CourseDAOImpl(FileDAL fileDAL, JsonParser jsonParser) {
   }
 
   @Override
-  public Course get(String name) {
-    Course res;
-    res = getAll().stream()
-        .filter(course -> course.getName().equals(name))
-        .findFirst()
-        .get();
+  public Course get(String name) throws PersistenceException {
+    try {
+
+      Optional<Course> res = getAll()
+          .stream()
+          .filter(course -> course.getName().equals(name))
+          .findFirst();
+      return res.isPresent() ? res.get() : null;
 
-    return res;
+    } catch (PersistenceException e) {
+      throw e;
+    }
   }
 
   @Override
-  public List<Course> getAll() {
+  public List<Course> getAll() throws PersistenceException {
     try {
       String json = fileDAL.readFile("Course.json");
       List<Course> courses = parser.toObjectList(json, Course[].class);
       return courses;
     } catch (ReadFileException e) {
-      e.printStackTrace(); //TODO throw model exceptions
+      throw new PersistenceException(e.getMessage());
```

Comment:
The mensage is not important, the important is what was the exception?

`new PersistanceException(e)` more better, just see the FileDAL